.github/workflows/release.main.kts

```diff
@@ -0,0 +1,134 @@
+import io.github.typesafegithub.workflows.actions.actions.SetupJava
+import io.github.typesafegithub.workflows.actions.gradle.ActionsSetupGradle
+import io.github.typesafegithub.workflows.actions.nexusactions.CreateNexusStagingRepo
+import io.github.typesafegithub.workflows.actions.nexusactions.DropNexusStagingRepoV1
+import io.github.typesafegithub.workflows.actions.nexusactions.ReleaseNexusStagingRepo
+import io.github.typesafegithub.workflows.domain.AbstractResult
+import io.github.typesafegithub.workflows.domain.JobOutputs
+import io.github.typesafegithub.workflows.domain.RunnerType
+import io.github.typesafegithub.workflows.domain.triggers.WorkflowDispatch
+import io.github.typesafegithub.workflows.dsl.expressions.Contexts
+import io.github.typesafegithub.workflows.dsl.expressions.expr
+import io.github.typesafegithub.workflows.dsl.workflow
+
+val SONATYPE_USERNAME by Contexts.secrets
+val SONATYPE_PASSWORD by Contexts.secrets
+val SONATYPE_STAGING_PROFILE_ID by Contexts.secrets
+val SIGNING_KEY_ID by Contexts.secrets
+val SIGNING_KEY by Contexts.secrets
+val SIGNING_PASSWORD by Contexts.secrets
+
+workflow(
+    name = "Publish release to Maven Central",
+    on = listOf(
+        WorkflowDispatch()
+    ),
+    sourceFile = __FILE__,
+) {
+    val stagingRepoJob =
+        job(
+            id = "create-staging-repo",
+            name = "Create staging repository",
+            runsOn = RunnerType.UbuntuLatest,
+            outputs = object : JobOutputs() {
+                var repositoryId: String by output()
+            }
+        ) {
+            val createRepo = uses(
+                action = CreateNexusStagingRepo(
+                    username = expr { SONATYPE_USERNAME },
+                    password = expr { SONATYPE_PASSWORD },
+                    stagingProfileId = expr { SONATYPE_STAGING_PROFILE_ID },
+                )
+            )
+            jobOutputs.repositoryId = createRepo.outputs.repositoryId
+        }
+    val publishJob =
+        job(
+            id = "publish-artifacts",
+            runsOn = RunnerType.MacOSLatest,
+            needs = listOf(stagingRepoJob),
+        ) {
+            uses(action = Checkout())
+            uses(
+                name = "Set up JDK",
+                action = SetupJava(
+                    javaVersion = "11",
+                    distribution = SetupJava.Distribution.Zulu,
+                    cache = SetupJava.BuildPlatform.Gradle,
+                ),
+            )
+            uses(
+                name = "Cache Kotlin Konan",
+                action = Cache(
+                    path = listOf(
+                        "~/.konan/**/*",
+                    ),
+                    key = "kotlin-konan-${expr { runner.os }}",
+                ),
+            )
+            uses(
+                name = "Set up Gradle",
+                action = ActionsSetupGradle(
+                    gradleVersion = "wrapper",
+                ),
+            )
```

Comment:
issue: Just realized that the caching won't work because the `build` workflow uses a different Gradle action. Do you mind changing it in the build workflow to the same one as here?

Reply:
Ah, good catch. Yes, I think `workflows
/build.main.kts` should be updated to be more like this workflow, which (I think) correctly follows the newer Gradle Actions set up.